src/discparse.py

```diff
@@ -217,30 +217,39 @@ async def get_dvdinfo(self, discs):
             path = each.get('path')
             os.chdir(path)
             files = glob(f"VTS_*.VOB")
-            files.sort()
-            filesdict = [[]]
+            # Switch to natural sorting
+            files = natsorted(files)
+            # Switch to ordered dictionary
+            filesdict = OrderedDict()
             main_set = []
-            for i in range(len(files)):
-                trimmed = files[i][4:]
-                set = int(trimmed[:2]) - 1
-                try:
-                    filesdict[set].append(trimmed)
-                except:
-                    filesdict.append([])
-                    filesdict[set].append(trimmed)
-            for vob_set in filesdict:
-                if len(vob_set) > len(main_set):
+            # Use ordered dictionary in place of list of lists
+            for file in files:
+                trimmed = file[4:]
+                if trimmed[:2] not in filesdict:
+                    filesdict[trimmed[:2]] = []
+                filesdict[trimmed[:2]].append(trimmed)
+                
+            for _, vob_set in filesdict.items():
+                # Parse media info for this VOB set
+                vob_set_mi = MediaInfo.parse(f"VTS_{vob_set[0][:2]}_0.IFO", output='JSON')
+                vob_set_mi = json.loads(vob_set_mi)
+                vob_set_duration = vob_set_mi['media']['track'][1]['Duration']
+                
+                # If no main set is identified yet, go ahead and define it and skip to next loop
+                if len(main_set) < 1:
                     main_set = vob_set
-                elif len(vob_set) == len(main_set):
-                    vob_set_mi = MediaInfo.parse(f"VTS_{vob_set[0][:2]}_0.IFO", output='JSON')
-                    vob_set_mi = json.loads(vob_set_mi)
-                    vob_set_duration = vob_set_mi['media']['track'][1]['Duration']
+                    continue
                     
-                    main_set_mi = MediaInfo.parse(f"VTS_{main_set[0][:2]}_0.IFO", output='JSON')
-                    main_set_mi = json.loads(main_set_mi)
-                    main_set_duration = main_set_mi['media']['track'][1]['Duration']
-                    if vob_set_duration > main_set_duration:
-                        main_set = vob_set
+                # Get media info for the existing main set
+                main_set_mi = MediaInfo.parse(f"VTS_{main_set[0][:2]}_0.IFO", output='JSON')
+                main_set_mi = json.loads(main_set_mi)
+                main_set_duration = main_set_mi['media']['track'][1]['Duration']
```

Comment:
while we're at it, it should be easy enough to use the vob_set info as the main_set info instead of re-parsing. I'll get to it this weekend if you haven't by then